src/utils/telemetry.ts

```diff
@@ -173,3 +213,213 @@ export function createTelemetry(
 		},
 	};
 }
+
+export function get_setup_ended_on_cli_setup_cancelled(): Events {
+	return {
+		name: "setup_ended",
+		payload: {
+			namespace: "onboarding",
+			steps: [
+				{
+					name: "emulator_installed",
+					is_first_step: true,
+					is_last_step: false,
+					step_order: 1,
+					status: "CANCELLED",
+				},
+				{
+					name: "auth_token_configured",
+					is_first_step: false,
+					is_last_step: false,
+					step_order: 2,
+					status: "SKIPPED",
+				},
+				{
+					name: "license_setup_ended",
+					is_first_step: false,
+					is_last_step: false,
+					step_order: 3,
+					status: "SKIPPED",
+				},
+				{
+					name: "aws_profile_configured",
+					is_first_step: false,
+					is_last_step: true,
+					step_order: 4,
+					status: "SKIPPED",
+				},
+			],
+			status: "CANCELLED",
+		},
+	};
+}
+
+export function get_setup_ended_on_authentication_cancelled(
+	authToken: string = "",
+): Events {
+	return {
+		name: "setup_ended",
+		payload: {
+			namespace: "onboarding",
+			steps: [
+				{
+					name: "emulator_installed",
+					is_first_step: true,
+					is_last_step: false,
+					step_order: 1,
+					status: "COMPLETED",
+				},
+				{
+					name: "auth_token_configured",
+					is_first_step: false,
+					is_last_step: false,
+					step_order: 2,
+					status: "CANCELLED",
+				},
+				{
+					name: "license_setup_ended",
+					is_first_step: false,
+					is_last_step: false,
+					step_order: 3,
+					status: "SKIPPED",
+				},
+				{
+					name: "aws_profile_configured",
+					is_first_step: false,
+					is_last_step: true,
+					step_order: 4,
+					status: "SKIPPED",
+				},
+			],
+			status: "CANCELLED",
+			auth_token: authToken,
+		},
+	};
+}
+
+export function get_setup_ended_on_license_setup_cancelled(
+	auth_token: string,
+): Events {
+	return {
+		name: "setup_ended",
+		payload: {
+			namespace: "onboarding",
+			steps: [
+				{
+					name: "emulator_installed",
+					is_first_step: true,
+					is_last_step: false,
+					step_order: 1,
+					status: "COMPLETED",
+				},
+				{
+					name: "auth_token_configured",
+					is_first_step: false,
+					is_last_step: false,
+					step_order: 2,
+					status: "COMPLETED",
+				},
+				{
+					name: "license_setup_ended",
+					is_first_step: false,
+					is_last_step: false,
+					step_order: 3,
+					status: "CANCELLED",
+				},
+				{
+					name: "aws_profile_configured",
+					is_first_step: false,
+					is_last_step: true,
+					step_order: 4,
+					status: "SKIPPED",
+				},
+			],
+			status: "CANCELLED",
+			auth_token: auth_token,
+		},
+	};
+}
+
+export function get_setup_ended_on_image_prefetch_cancelled(
+	auth_token: string,
+): Events {
+	return {
+		name: "setup_ended",
+		payload: {
+			namespace: "onboarding",
+			steps: [
+				{
+					name: "emulator_installed",
+					is_first_step: true,
+					is_last_step: false,
+					step_order: 1,
+					status: "CANCELLED",
```

Comment:
This is a tricky one! Prefetch doesn't have a separate step defined in tracking. However, it is also not part of the install. For example, one can install the emulator but then cancel the pre-fetch. I didn't define pre-fetch as a new separate step because it runs in the background and it's hard to define the order. Options I see:
- report all previous steps as completed/skipped but overall status as cancelled
- add a separate step for pre-fetch.

Which one would you prefer?
Just in case, here is suggestion for the first option:

```suggestion
					status: "COMPLETED",
```

Reply:
I was unsure how to capture it. You're right, it doesn't fall inside emulator_installed.
We can go for the simpler version for now, the first option you proposed.